internal/events/aggregator.go

```diff
@@ -476,14 +476,10 @@ func (ag *aggregator) processMessage(ctx context.Context, manifest *core.BatchMa
 
 	}
 
-	dispatched := false
-	var newState core.MessageState
-	if dataAvailable {
```

Comment:
I removed this check because there's no way to get this far if data was missing.

Reply:
@awrichar - why do we *not* need to call `state.SetContextBlockedBy(ctx, *unmaskedContext, pin.Sequence)` in the case of `!dataAvailable`

That change wasn't made in this PR, and I'm not paged back in enough to answer this myself. But I read the comments on the `if dispatched {` block below and it strikes me it applies on face value:

```
	// - dispatched=false: we need to prevent dispatch of any subsequent messages on the same topic in the batch
```

Reply:
That question also occurred to me, and I was somewhat hoping you knew the answer 😅 sounds like one or both of us needs to page back in more fully on the difference between waiting, rejecting, and blocking.

Reply:
After looking again, it does feel like we should mark the context blocked when the message/data is unavailable. I rearranged the code blocks so it will fall through to that path now.